packages/keystone/src/lib/core/field-assertions.ts

```diff
@@ -87,12 +87,15 @@ function assertIdFieldGraphQLTypesCorrect(list: ListForValidation) {
       `The idField on a list must not define an update GraphQL input but the idField for ${list.listKey} does define one`
     );
   }
-  if (idField.access.read === false) {
+  if (idField.graphql.isEnabled.read === false) {
     throw new Error(
-      `The idField on a list must not have access.read be set to false but ${list.listKey} does`
+      `The idField on a list must not have graphql.isEnabled.read be set to false but ${list.listKey} does`
     );
   }
-  if (idField.output.type.kind !== 'non-null' || idField.output.type.of !== graphql.ID) {
+  if (
+    idField.output &&
```

Comment:
```suggestion
    !idField.output ||
```

Reply:
Seems like this change was reverted?
```suggestion
    !idField.output ||
```

Reply:
Ah yep, though this new check isn't actually needed anymore, since the change to output has been reverted, so I'm just gonna set this back to what it was originally.